gossip/src/cluster_info.rs

```diff
@@ -564,7 +564,10 @@ impl ClusterInfo {
                     //    definition that information is already available
                     let contact_info = v.value.contact_info().unwrap();
                     if contact_info.id != self_pubkey
-                        && !entrypoint_gossip_addrs.contains(&contact_info.gossip)
+                        && match contact_info.gossip() {
+                            Ok(addr) => !entrypoint_gossip_addrs.contains(&addr),
+                            Err(_) => false,
+                        }
```

Comment:
nit: I think you did something else similar in this PR, but I think you could do this with combinators and avoid the match if you're so inclined:
```rust
&& contact_info.gossip().ok().map(|addr| !entrypoint_gossip_addrs.contains(&addr)).unwrap_or_default()
```

Reply:
done.